shopview.py

```diff
@@ -1,34 +1,96 @@
 import arcade
 import const.constants as C
+import os
 from lib import global_scale
 
 # Base ShopView
 
 
 class ShopView(arcade.View):
-
     def __init__(self):
-
+        # Inherit parent class
         super().__init__()
 
+        self.cursor_sprite = None
+        self.Player1 = None
+        self.list1 = os.listdir("assets/")
+
+        # Create the sprite lists
+        self.move_spritel = arcade.Sprite(
+            "resources/images/shop/shop_left.png", 0.2)
+        self.move_spriter = arcade.Sprite(
+            "resources/images/shop/shop_right.png", 0.2)
+        self.move_spritel.center_x = 100
+        self.move_spritel.center_y = 350
+        self.move_spriter.center_x = 1200
+        self.move_spriter.center_y = 350
+        self.cursor_sprite = arcade.Sprite(
+            "resources/images/goat_cursor.png", 1)
+        self.c_p = 0
+        self.Player1 = PreviewSprite(self.list1[self.c_p])
+
+    def on_draw(self):
+        """Render the screen."""
+
+        # Clear the screen to the background color
+        self.clear()
+
+        # arcade.draw_lrwh_rectangle_textured(0, 0,
+        #                                     C.SCREEN_WIDTH, C.SCREEN_HEIGHT,
+        #                                     self.background)
+
+        self.move_spritel.draw(pixelated=True)
+        self.move_spriter.draw(pixelated=True)
+        self.cursor_sprite.draw()
+        self.Player1.draw()
+
     def on_show(self):
         """Called when switching to this view."""
         arcade.set_background_color(C.MENU_BACKGROUND_COLOR)
 
-    def on_draw(self):
-        """Draw the menu"""
-        self.clear()
+    def on_mouse_press(self, x, y, button, modifiers):
+
+        if self.move_spritel.collides_with_sprite(self.cursor_sprite):
+            self.c_p = (self.c_p + 1)
+            if self.c_p > len(self.list1):
+                self.c_p = 0
+            self.Player1 = PreviewSprite(self.list1[self.c_p])
```

Comment:
this line causes an error
![](https://cdn.discordapp.com/attachments/953179134396358659/957206900473233478/unknown.png)
in certain index / image
![](https://cdn.discordapp.com/attachments/953179134396358659/957206933218160700/unknown.png)

Reply:
Fixed in 6e4f722eae0c53c23b8a5e76561c856bcc557b93